src/python/pants/core/util_rules/strip_source_roots.py

```diff
@@ -93,20 +96,22 @@ def determine_source_root(path: str) -> str:
         return PurePath(path).parent.as_posix()
 
     if request.representative_path is not None:
-        resulting_digest = await Get[Digest](
-            RemovePrefix(
-                request.snapshot.digest, determine_source_root(request.representative_path),
-            )
-        )
-        resulting_snapshot = await Get[Snapshot](Digest, resulting_digest)
-        return SourceRootStrippedSources(snapshot=resulting_snapshot)
+        source_root = determine_source_root(request.representative_path)
```

Comment:
Still might be worth checking for `source_root == ""`? Mostly so superfluous prefix removal work doesn't show up in the build output.

Reply:
I agree. I plan to do this in a followup.